macOS homebrew `libxmlsec1` 1.3.3 incompatible with `samael`

```
   Compiling samael v0.0.14
error[E0432]: unresolved import `crate::bindings::xmlSecOpenSSLAppKeyLoad`
  --> /Users/david/.cargo/registry/src/index.crates.io-6f17d22bba15001f/samael-0.0.14/src/xmlsec/backend.rs:14:9
   |
14 |         xmlSecOpenSSLAppKeyLoad as xmlSecCryptoAppKeyLoad,
   |         -----------------------^^^^^^^^^^^^^^^^^^^^^^^^^^
   |         |
   |         no `xmlSecOpenSSLAppKeyLoad` in `bindings`
   |         help: a similar name exists in the module: `xmlSecOpenSSLAppKeyLoadEx`

   Compiling mg-admin-client v0.1.0 (/Users/david/oxide/omicron/clients/mg-admin-client)
   Compiling dpd-client v0.1.0 (/Users/david/oxide/omicron/clients/dpd-client)
For more information about this error, try `rustc --explain E0432`.
error: could not compile `samael` (lib) due to previous error
```

[xmlsec](https://www.aleksey.com/xmlsec/) 1.3.3 came out on 2024-01-04, but `libxmlsec1` on homebrew was only bumped from 1.3.2 to 1.3.3 [yesterday](https://github.com/Homebrew/homebrew-core/commits/f5c6266aaf51084a7f9e5d1917cb8f566fee0395/Formula/lib/libxmlsec1.rb). The samael dep in omicron was bumped to 0.0.14 [five days ago](https://github.com/oxidecomputer/omicron/commit/e261a960cb365ad92f103a35b262713118ea6441), and I'm pretty sure I had #4903 building locally on top of that, so I believe this is primarily about `libxmlsec1`.

There seems to be no way to force homebrew to install `1.3.2` to test the problem. However, I went into `/Users/david/.cargo/registry/src/index.crates.io-6f17d22bba15001f/samael-0.0.14/src/xmlsec/backend.rs` and changed the [offending line](https://github.com/njaremko/samael/blob/4b4811d97ae324da1376624e77e82af18ee108f3/src/xmlsec/backend.rs#L14) to reference `xmlSecOpenSSLAppKeyLoadEx` (I know this is not good) and it did let it compile and let me run an integration test (that does not use SAML).

I was able to find the spot where `xmlSecOpenSSLAppKeyLoad` was deleted from xmlsec:

https://github.com/lsh123/xmlsec/commit/114f580409e076bbe7b352eb74008d818c6e3e93#diff-d2b6e591794d60dda4a23d8bd4a565c169ee7c9ef724d050692c1518af6eeefdL89

`xmlSecOpenSSLAppKeyLoadEx` seems to have a different signature, so I'm guessing that even though it compiles, it would blow up at runtime if I actually exercised this function. It seems to me the real fix would be in the import in `samael`.

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -32,14 +32,6 @@ jobs:
     - name: Disable packages.microsoft.com repo
       if: ${{ startsWith(matrix.os, 'ubuntu') }}
       run: sudo rm -f /etc/apt/sources.list.d/microsoft-prod.list
-    # https://github.com/oxidecomputer/omicron/issues/4920
-    - name: Pin libxmlsec1 to 1.3.2
-      if: ${{ startsWith(matrix.os, 'macos') }}
-      run: |
-        curl -fLOsS --retry 5 https://raw.githubusercontent.com/Homebrew/homebrew-core/081149b0d2720c2759b6ac8253e33b27f6d6c1cd/Formula/lib/libxmlsec1.rb
-        brew install ./libxmlsec1.rb
-        brew pin libxmlsec1
-        rm -f libxmlsec1.rb
     - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
       with:
         ref: ${{ github.event.pull_request.head.sha }} # see omicron#4461

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -614,3 +614,8 @@ branch = "oxide/omicron"
 # to it.
 [patch.crates-io.omicron-workspace-hack]
 path = "workspace-hack"
+
+# Pulls in https://github.com/njaremko/samael/pull/41
+[patch.crates-io.samael]
+git = "https://github.com/oxidecomputer/samael"
+branch = "oxide/omicron"

